### DISCUSSION TITLE
httpx.post get a typing error when data is plain string

### DISCUSSION THREAD

**Discussion by szw0407** [encode/httpx#2849] [Potential Issue] [answered]
Usually http post method sends data in JSON format, and the `data` in `httpx.post` is assigned to as `typing.Optional[RequestData]` type. However under some certain circumstances, some APIs require post data in other formats. 

I used `httpx.post` to get data from an API, which requires a plain text in a certain format, but neither json nor xml, as the post body.

Yes, that is quite rare, and I suppose such APIs are badly designed. However, since `content-type` in `header` could be `text\plain` as well as other values, the `RequestData` type should not be simply `Mapping[str, Any]`.

I get a typing error saying str is incompatible with RequestData when I write such codes:
```python
import httpx

def example function(url:str, data_str:str):
    return httpx.post(url, data=data_str, header = {"Content-Type":"text/plain"})
```

Anyway, it works properly when I disable typing check. However, I wish the typing could be reconsidered.

I simply changed the line `RequestData = Union[Mapping[str, Any], Sequence[Tuple[str, Any]], bytes, str, None]`, but I guess that is not proper. I wish it could be checked based upon the `Content-Type` value in the header.

**Comment by karpetrosyan** [comment, marked as answer]
Hi! I think the content argument is what you are searching for.

You can do something like this

```python
httpx.post(url, content="my plain text", header = {"Content-Type":"text/plain"})
```
